src/app/ReadClient.h

```diff
@@ -486,7 +486,7 @@ class ReadClient : public Messaging::ExchangeDelegate
     void CancelResubscribeTimer();
     void MoveToState(const ClientState aTargetState);
     CHIP_ERROR ProcessAttributePath(AttributePathIB::Parser & aAttributePath, ConcreteDataAttributePath & aClusterInfo);
-    CHIP_ERROR ProcessReportData(System::PacketBufferHandle && aPayload);
+    CHIP_ERROR ProcessReportData(System::PacketBufferHandle && aPayload, bool aIsUnsolicitedReport);
```

Comment:
Since we have the `/* aIsUnsolicitedReport */` everywhere this kind of hints that a BOOL argument is not ideal. 

lets use an `enum class` even if it is practically just two values, then we can clearly read `ReportType::kUnsolicited` and similar.

Reply:
@andy31415 Done.